### PR TITLE
INTYGFV-15361: Added sjf within same unit

### PR DESCRIPTION
**Pull request by mhornfeldt**

### CODE REVIEW

**Comment by mhornfeldt** [comment]
I found the translation from Sammanhållen journalföring to "National Patient Overview" at KI. Should we go for that or does someone have a better suggestion? @ChrilleSigerhed @SamuelTilly @mikeyOne

File: apps/rehabstod/src/pages/Patient/components/patientOverview/BlockedInformationCard.test.tsx
Comment (SamuelTilly) [inline apps/rehabstod/src/pages/Patient/components/patientOverview/BlockedInformationCard.test.tsx:1]: This component holds no logic itself, everything is already tested with `BlockedInformation`, `PatientOverviewCard` and `ExpandableCard`.

File: apps/rehabstod/src/pages/Patient/Patient.tsx [hunk 1]
```diff
@@ -24,13 +25,13 @@ export function Patient() {
...
           </>
         )}
-
         {earlierSickLeaves.length > 0 && (
           <>
             <h2 className="ids-heading-2 text-neutral-20">Tidigare sjukfall på {user?.valdVardenhet?.namn}</h2>
             <PatientSickLeaves sickLeaves={earlierSickLeaves} />
           </>
         )}
+        <PatientOverview sjfMetaData={patient?.sjfMetaData} patientId={patient ? patient.sjukfallList[0].intyg[0].patient.id : ''} />
```
Comment (SamuelTilly) [inline apps/rehabstod/src/pages/Patient/Patient.tsx:34]: Could `patientId` be set to `atob(patientId)`?

File: apps/rehabstod/src/components/SickLeave/SickLeaveDegreeInfo.tsx [hunk 2]
```diff
@@ -3,15 +3,21 @@ import { IDSIcon } from '@frontend/ids-react-ts'
 export function SickLeaveDegreeInfo({ degrees }: { degrees: number[] }) {
   const hasSeveralDegrees = degrees.length > 1
 
+  if (degrees.length === 0) {
+    return <td>Okänt</td>
```
Comment (SamuelTilly) [inline apps/rehabstod/src/components/SickLeave/SickLeaveDegreeInfo.tsx:7]: This converts the component to a table cell, perhaps we could just use `<span>Okänt</span>` instead.

File: apps/rehabstod/src/pages/Patient/components/patientOverview/OpenInformationCard.test.tsx
Comment (SamuelTilly) [inline apps/rehabstod/src/pages/Patient/components/patientOverview/OpenInformationCard.test.tsx:1]: This component holds no logic itself, everything is tested with `PatientOverviewCard` and `ExpandableCard`. `OpenInformation` could be moved in to reduce the number of components, same as `BlockedInformation`.

File: apps/rehabstod/src/pages/Patient/components/patientOverview/PatientOverview.tsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        subTitle="Vårdenhet att hämta information från"
+        description="Det finns ospärrad information hos en annan vårdenhet inom din vårdgivare. Du kan klicka nedan för att visa vilka vårdenheter som
+        har denna information och få möjlighet att inhämta den."
+        items={sjfMetaData.kraverInteSamtycke}
+        onGetInformation={handleGetCareUnitInformation}
+      />
+      <BlockedInformationCard
+        title="Spärrad information inom vårdgivare"
+        subTitle="Vårdenheter"
+        description="Det finns spärrad information hos en annan vårdenhet inom din vårdgivare. Endast patienten kan få spärren hävd genom att kontakta den enhet där spärren sattes. Du kan klicka nedan för att visa vilka vårdenheter som har spärrad information hos sig."
```
Comment (SamuelTilly) [inline apps/rehabstod/src/pages/Patient/components/patientOverview/PatientOverview.tsx:30]: These attributes are unique for; and could be moved to `BlockedInformationCard`

File: apps/rehabstod/src/pages/Patient/components/patientOverview/PatientOverview.tsx [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    addUnit({ patientId, vardenhetId: id })
+  }
+
+  return (
+    <div className="flex flex-col gap-5">
+      <OpenInformationCard
+        title="Ospärrad information inom vårdgivare"
+        subTitle="Vårdenhet att hämta information från"
+        description="Det finns ospärrad information hos en annan vårdenhet inom din vårdgivare. Du kan klicka nedan för att visa vilka vårdenheter som
+        har denna information och få möjlighet att inhämta den."
```
Comment (SamuelTilly) [inline apps/rehabstod/src/pages/Patient/components/patientOverview/PatientOverview.tsx:23]: These attributes are unique for; and could be moved to `OpenInformationCard`

File: apps/rehabstod/src/pages/Patient/components/patientOverview/PatientOverview.tsx [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+        items={sjfMetaData.kraverInteSamtycke}
+        onGetInformation={handleGetCareUnitInformation}
```
Comment (SamuelTilly) [inline apps/rehabstod/src/pages/Patient/components/patientOverview/PatientOverview.tsx:25]: `useAddVardenhetMutation` could be moved to `OpenInformationCard`

**Review by SamuelTilly — CHANGES_REQUESTED** [review]
Looking good!, only some minor suggestions.

File: apps/rehabstod/src/pages/Patient/components/patientOverview/PatientOverview.tsx (hunk 5 above)
Comment (mhornfeldt) [reply apps/rehabstod/src/pages/Patient/components/patientOverview/PatientOverview.tsx:25]: I left it out to maybe be able to reuse code when implementing the second jira. I'll leave it out and if possible make the refactor in the next jira.

File: apps/rehabstod/src/pages/Patient/components/patientOverview/PatientOverview.tsx (hunk 4 above)
Comment (mhornfeldt) [reply apps/rehabstod/src/pages/Patient/components/patientOverview/PatientOverview.tsx:23]: I'll see if I can move it in the next jira but I think for reuse purposes it should stay like this for now.

File: apps/rehabstod/src/pages/Patient/components/patientOverview/PatientOverview.tsx (hunk 3 above)
Comment (mhornfeldt) [reply apps/rehabstod/src/pages/Patient/components/patientOverview/PatientOverview.tsx:30]: I'll see if I can move it in the next jira but I think for reuse purposes it should stay like this for now.

**Pull request merged**